### PR TITLE
Fix Nemotron 3 SFT example: add Mamba modules to LoRA targets

### LINKED ISSUE #6773: Nemotron 3 SFT example uses target_modules that may not exist in hybrid architecture
## Problem

TRL's official Nemotron 3 SFT example (`examples/scripts/sft_nemotron_3.py`) uses LoRA target modules that may not exist in NemotronH's hybrid architecture:

```python
target_modules=["q_proj", "k_proj", "v_proj", "o_proj", 
                "gate_proj", "up_proj", "down_proj"]
```

The `gate_proj`, `up_proj`, `down_proj` modules are standard MLP projection names from Llama-style architectures. NemotronH uses Mamba-2 layers (with `in_proj`, `out_proj`) and MoE layers (with different naming). These MLP module names may silently not match anything in the model.

## Impact

Users following the official example may get LoRA adapters that only target a small fraction of the model's layers (the attention projections), with the MLP/Mamba modules silently skipped. This produces the same "loss not improving" failure mode we experienced:

- DPO LoRA training: 25 steps, loss stuck at 0.6931 (ln(2))
- Zero behavioral change in the model
- No warning that target modules didn't match

## Reproduction

We ran DPO LoRA on `nvidia/NVIDIA-Nemotron-Nano-9B-v2` with attention-only target modules (same as the example minus the MLP modules). The LoRA adapter only touched 4 out of 56 layers (7.1%). Loss never moved from ln(2). Full writeup: https://github.com/davidnichols-ops/claude-yolo-vibes-v5/blob/master/EVALUATION.md

## Proposed Fix

1. Update the example to use target modules that actually exist in NemotronH:

```python
# Inspect the model to find valid target modules
import re
target_modules = list(set(
    name.split(".")[-1] for name, module in model.named_modules()
    if isinstance(module, torch.nn.Linear) and 
    re.match(r".*\d+\..*", name)  # only layer-level modules
))
```

Or explicitly list NemotronH-appropriate modules:

```python
target_modules=[
    # Attention
    "q_proj", "k_proj", "v_proj", "o_proj",
    # Mamba (if PEFT allows — see PEFT issue about mamba_model_types ban)
    "in_proj", "out_proj", 
    # MoE/MLP — check actual module names in the model
    # "gate", "up", "down"  # verify these exist before including
]
```

2. Add a comment in the example explaining that NemotronH is a hybrid architecture and users should verify target modules with `model.named_modules()`.

3. Consider adding a runtime check in TRL that warns when configured `target_modules` don't match any modules in the model.

## Environment

- trl: 1.10.0
- transformers: 5.16.0.dev0
- Model: nvidia/NVIDIA-Nemotron-Nano-9B-v2

## Related

- PEFT PR #3289: Added default target_modules for nemotron_h (attention only)
- PEFT issue (filed separately): NemotronH LoRA target modules silent failure
- Full writeup: https://github.com/davidnichols-ops/claude-yolo-vibes-v5/blob/master/EVALUATION.md

### PR DESCRIPTION
**Pull request by davidnichols-ops**
# What does this PR do?

Fixes the Nemotron 3 SFT example (`examples/scripts/sft_nemotron_3.py`) which had incorrect LoRA target modules for the hybrid Mamba-Attention architecture.

Fixes #6773

## Problem

NemotronH is a hybrid Mamba-Attention architecture. The previous `--lora_target_modules` in the example docstring were:

```
q_proj k_proj v_proj o_proj gate_proj up_proj down_proj
```

Two issues:

1. **`gate_proj` does not exist** in NemotronH. The MLP layers use `up_proj`/`down_proj`, not `gate_proj`.
2. **Mamba layers were missing entirely.** NemotronH models have Mamba SSM layers with `in_proj`/`out_proj` modules. In Nemotron-Nano-9B-v2, 52 of 56 layers are Mamba. Omitting these from LoRA targets means 93% of the model is frozen during fine-tuning, resulting in zero learning (DPO loss stays at ln(2), HumanEval drops to 0%).

See https://github.com/huggingface/peft/issues/3554 for the full analysis.

## Fix

Changed `--lora_target_modules` from `q_proj k_proj v_proj o_proj gate_proj up_proj down_proj` to `q_proj k_proj v_proj o_proj in_proj out_proj up_proj down_proj`, and added an explanatory comment about the hybrid architecture.

## Before submitting

- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request), Pull Request section?
- [x] Was this discussed/approved via a GitHub issue? Please add a link to it if that's the case. (Issue #6773)
- [x] Did you make sure to update the documentation with your changes? (Added explanatory comment in the docstring)
- [ ] Did you write any new necessary tests? (N/A — this is an example script fix)

## AI writing disclosure

- [x] AI-assisted: some parts were suggested or improved by AI, but the PR was written and reviewed by a human.

## Who can review?

Anyone familiar with NemotronH or LoRA fine-tuning examples.

### CODE REVIEW

**Comment by davidnichols-ops** [comment]
Thanks for the quick judgment. you keep rockin that banner hammer lmao

**Comment by qgallouedec** [comment]
Thanks for the follow-up.

For transparency, let me explain what happened.

We receive an insane amount of sloppy AI contributions, which has become a broader OSS problem with the recent improvements in coding agents.

Consequently, to avoid drowning maintainers in these PRs, solving artificial issues (do you maintain an OSS project yourself?), we have to filter them automatically and aggressively based on heuristics, unfortunately preferring false positives.

In this case, you opened https://github.com/huggingface/trl/pull/6775 without following the PR template, thus not following the contribution guide. a typical pattern we see with sloppy contributions, so your profile was categorized accordingly.

If this was a mistake, please reopen the PR.

**Comment by davidnichols-ops** [comment]
Also, I really do like the architecture here, but I WILL NOT contribute or waste my time with slap-happy ban hammers made by overwhelmed ML engineers. I will, on second thought, close my revised PR, as your team is incapable of handling it at this moment.
